examples/research_projects/codeparrot/scripts/preprocessing.py

```diff
@@ -49,18 +49,77 @@ def is_autogenerated(example, scan_width=5):
         return {"autogenerated": False}
 
 
+def is_config_test(example, scan_width=5, max_config=4, max_test=4, max_len=8000):
+    """Check if file is a configuration file or a unit test by :
+         1- looking for keywords in the first few lines of the file
+         2- counting number of occurence of the words 'config' and 'test'.
+    Remark: for the second test we only check files with less than 8000 characters"""
+
+    keywords = ["unit tests", "test file", "configuration file"]
+    lines = example["content"].splitlines()
+    count_config = 0
+    count_test = 0
+    # first test
+    for _, line in zip(range(scan_width), lines):
+        for keyword in keywords:
+            if keyword in line.lower():
+                return {"config_test": True}
+    # second test
+    if len(example["content"]) < max_len:
+        for line in lines:
+            count_config += line.lower().count("config")
+            count_test += line.lower().count("test")
+            if count_config > max_config or count_test > max_test:
+                return {"config_test": True}
+    return {"config_test": False}
+
+
+def is_uncommon(example):
```

Comment:
maybe we could rename this to `has_no_keywords` (same for returned dict) for easier readability.